### PR TITLE
Kong example application

### PR DESCRIPTION
**Pull request by siddharthist**
Fixes: #484 
Tested on DigitalOcean

### CODE REVIEW

**Comment by sinzone** [comment]
+1

**Comment by siddharthist** [comment]
@sinzone Were you able to test this/get it working?

**Comment by siddharthist** [comment]
@stevendborrelli What would be the best way to make this more dynamic? Consul-template? I guess I just didn't know enough to do so :) 

@sinzone I'd love to be forwards-compatible, maybe we can use a container based on the RC1 or something. Could you point me towards the new configuration documentation?

**Comment by stevendborrelli** [comment]
It looks like there are only 2-3 fields that are updated. You could do something like have a new container that runs sed to replace via env variables, or use consul-template with defaults.

**Comment by thibaultCha** [comment]
[This document](https://github.com/Mashape/kong/blob/next/UPGRADE.md#upgrade-to-06x) shows which fields were updated and how. It does not show the new ones.

The new configuration file looks like [this](https://github.com/Mashape/kong/blob/feat/invalidations/kong.yml) and the new fields are there. All properties in there are documented and omittable because the commented values are the default ones.

**Comment by ryane** [comment]
@siddharthist this is really cool. I'm going to post an update to this branch with some changes to workaround some problems I ran into.

**Comment by ryane** [comment]
This uses the "uris" option in the marathon.json. This saves us from having to run ansible and copying the config to all workers. It does mean you have to have your config in an accessible location (you could also probably setup a marathon artifact store).

The Kong proxy layer is now load-balanced by traefik by default.

I was getting a lot of failures when using a multi-node cassandra cluster. Kong would continually thrash between successful requests and returning `An unexpected error occurred` in the json responses. The health checks would fail and Marathon would keep restarting Kong. In the kong error logs, I was seeing things like: `2016/01/15 18:28:26 [error] 60#0: *1 [lua] responses.lua:61: Cassandra returned error (Unprepared): Prepared query with ID c5bd483fc757a66975a3c297d62407e7 not found (either the query was not prepared on this host (maybe the host has been restarted?) or you have prepared too many queries and it has been evicted from the internal cache)`. I don't know enough about Cassandra to understand exactly what the problem is but it did go away when I individually listed each node in the `contact_points` configuration.

```
- "10.0.0.4:9042"
- "10.0.0.6:9042"
- "10.0.0.7:9042"
```

For now, I updated the script to install a single node cassandra cluster but this is probably not ideal. To support a multi-node cluster, it looks like we will need something like consul-template to write out each individual cassandra node. Unless someone knows how to fix it so that it works when using the single consul domain...

I wasn't having much luck with the 0.6 RC2 release (I was trying a bunch of things and didn't capture what the problem was) but it might be worth another look.

File: examples/kong/kong.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
+    "id": "/kong",
+    "dependencies": [],
+    "instances": 1,
+    "cpus": 1,
+    "mem": 1024,
+    "cmd": "cp $MESOS_SANDBOX/kong.yml /etc/kong/; kong start && tail -f /usr/local/kong/logs/error.log",
+    "uris": [
+      "http://ryanesc.s3.amazonaws.com/samples/kong.yml"
```
Comment (ryane) [inline examples/kong/kong.json]: For now, this is in my personal S3 account. If we like this approach better, we should move it to a more "official" location.

**Comment by ryane** [comment]
One more update to run kong 0.6rc3 which fixes the cassandra issues described above. see https://github.com/Mashape/kong/issues/886.

**Comment by siddharthist** [comment]
LGTM, with good bintray link. Just needs final testing. Thanks for the hard work @ryane!

**Comment by TanyaCouture** [comment]
Successfully deployed with OS:
    ./deploy.sh run
    Kong and Cassandra healthy MarathonUI
    Kong ports were 31839,31840
    Pinging 31840 produce successful json(ansible all -i plugins/inventory/terraform.py -a "curl kong.service.consul:31840")

**Comment by sinzone** [comment]
@siddharthist great job!  

@ryane We're releasing 0.6 tmrw. 

Also worth updating readme and website?

**Comment by siddharthist** [comment]
We should test once and update to 0.6. Shouldn't be any more issues, and it should improve stability for those who are using this example.

**Comment by sinzone** [comment]
https://github.com/Mashape/kong/releases/tag/0.6.0

**Comment by ryane** [comment]
thanks see #1071

**Pull request merged**